youtube_dl/YoutubeDL.py

```diff
@@ -1291,6 +1289,22 @@ def process_info(self, info_dict):
             self.report_error('unable to create directory ' + compat_str(err))
             return
 
+        filesystem_enc = sys.getfilesystemencoding()
+        if filesystem_enc is None:
+            filesystem_enc = 'utf-8'
+        max_filename_len = 200
+        if sys.platform.startswith('linux'):
```

Comment:
os.pathconf and os.pathconf_names both available on unix and i use them only inside the condition to check for linux and because i can't test it on mac os i didn't add it.
and about test if os.pathconf_names contains PC_NAME_MAX i will add it now.
and i want to ask when i decode the bytes should i let python choose the encoding or use the same filesystem encoding that encode the string with.
```python
info_dict['title'] = info_dict['title'].encode(filesystem_enc)[:trun_len].decode() + '...'
```
or
```python
info_dict['title'] = info_dict['title'].encode(filesystem_enc)[:trun_len].decode(filesystem) + '...'
```